components/eamxx/src/physics/mam/eamxx_mam_generic_process_interface.cpp

```diff
@@ -342,7 +347,13 @@ void MAMGenericInterface::add_tracers_wet_atm() {
   add_tracer<Required>("qi", grid_, q_unit);
 
   // cloud liquid number mixing ratio [1/kg]
-  add_tracer<Required>("nc", grid_, n_unit);
+  // FIXME: THE FOLLOWING IS A HACK
+  // We need this to allow nc to be mixed in mam (via nc_tend) but not shoc
+  if(use_dynamics_advected_only_nc_) {
+    add_tracer<Required>("nc", grid_, n_unit, 1, TracerAdvection::DynamicsOnly);
+  } else {
+    add_tracer<Required>("nc", grid_, n_unit);
+  }
```

Comment:
Is it correct that this portion is not a hack, but that the hack is only in P3 where we need to change the `nc` advection type? So MAM will still have this decision to make which will be controled by a process yaml option? Or does MAM always want `nc` not advected by shoc?

If MAM will always want the choice about `nc`, I would say we should setup MAM in this PR to look like it will when tracer requests are fixed, which I think just requires removing the "hacked" language from the comments here (and above when querying the parameter and below in the .hpp).